SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1300,9 +1301,31 @@ void USpatialActorChannel::ServerProcessOwnershipChange()
 		}
 	}
 
-	UpdateEntityACLToNewOwner();
+	// Changing an Actor's owner can affect its NetConnection so we need to reevaluate this.
+	FString NewClientConnectionWorkerId;
+	{
+		SCOPE_CYCLE_COUNTER(STAT_GetOwningClientWorkerId);
+		NewClientConnectionWorkerId = SpatialGDK::GetOwningClientWorkerId(Actor);
```

Comment:
Assuming it's valid for this string to be null. 
Also wondering if it's possible for the new worker to be the same as the old worker, in which case we could early out.

Reply:
Yep good call. Gonna do another little optimisation here. We're doing a big hierarchy iteration here which we might not need.

We need to iterate through children and update IFF the connection owning worker ID changes OR the interest bucket changes.

Currently, we iterate through every Actor in the hierarchy and check both, the worker ID is a string comparison, however the interest bucket change is much less efficient.

Changes:
- Save current interest component bucket - same as what is done for connection id
- Only send component presence / entity ACL updates if worker owning connection worker ID changes
- Only iterate through child Actors if one of the owning connection worker ID OR interest bucket changes